sktime/base/_base.py

```diff
@@ -73,6 +73,9 @@ class BaseObject(_BaseObject):
     Extends skbase BaseObject with additional features.
     """
 
+    # global dependency alias tag for sklearn dependency management
+    _tags = {"python_dependencies_alias": {"scikit-learn": "sklearn"}}
+
```

Comment:
I suggest placing the _tags  for python_dependencies_alias in `BaseEstimator` instead of `BaseObject`, since the tagging is designed for estimators anyway. Also, based on the test created in *test_base.py*, `BaseObject` requires a clean tag (i.e., a class with no tag at all) for all test cases to succeed. If you still want to place the tag in `BaseObject`, you may need to directly modify the test fixtures directly.

Reply:
A small request from me 😅 (ignore this if somehow burdens you - I don't mind 😇), if you think the suggestion is great and place the _tag in `BaseEstimator` later, could you make the [commit as *commit  with coauthors*](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors). will be great to collect [github achievement badges](https://github.com/drknzz/GitHub-Achievements) here. hehe

Reply:
hm, never done that manually - how about you PR to this branch, that would count, right?

(agree with moving tag to `BaseEstimator`)

Reply:
> how about you PR to this branch, that would count for the badge once we merge this branch to `main`, right?

not quite sure but I think it's worth trying - at least get to learn to make a PR to a different branch other than `main`

Reply:
(I think it counts because there are commits from different people in the branch - they get squashed to a single commit to `main` authored by multiple contributors)

Reply:
alright, thanks!! :)